Sources/Components/Radio/Radio.swift

```diff
@@ -0,0 +1,136 @@
+        ///   - state: The state of the radio button group (default, error, disabled).
+        public init(selectedOption: Binding<Option>,
+                    options: [Option],
+                    label: @escaping (Option) -> String,
+                    state: RadioButtonState = .default) {
+            self._selectedOption = selectedOption
+            self.options = options
+            self.label = label
+            self.state = state
+        }
+        
+        public var body: some View {
+            VStack(alignment: .leading, spacing: Spacing.spacing200) {
+                ForEach(options) { option in
+                    RadioButton(isSelected: selectedOption == option,
+                                label: label(option),
+                                state: state) {
+                        if state != .disabled {
+                            selectedOption = option
+                        }
+                    }
+                    .disabled(state == .disabled)
+                }
+            }
+            .padding(Spacing.spacing200)
+        }
+    }
+    
+    /// A single radio button view used within `RadioButtonGroup`.
+    ///
+    /// Displays a selectable circular button with a label. When the button is selected,
+    /// the circle is filled to indicate the selection.
+    ///
+    /// - Parameters:
+    ///   - isSelected: A Boolean value indicating whether the radio button is selected.
+    ///   - label: The text label for the radio button.
+    ///   - state: The state of the radio button (default, error, disabled).
+    ///   - action: A closure that is executed when the radio button is tapped.
+    struct RadioButton: View {
+        /// A Boolean value indicating whether the radio button is selected.
+        var isSelected: Bool
+        /// The text label for the radio button.
+        var label: String
+        /// The state of the radio button (default, error, disabled).
+        var state: RadioButtonState
+        /// A closure that is executed when the radio button is tapped.
+        var action: () -> Void
+        /// Object that will provide needed colors.
+        private let colorProvider: ColorProvider = Warp.Color
+        
+        var body: some View {
+            HStack(spacing: Spacing.spacing100) {
+                Circle()
+                    .strokeBorder(borderColor, lineWidth: isSelected ? 6 : 1)
+                    .background(Circle().fill(fillColor))
+                    .frame(width: 20, height: 20)
+                SwiftUI.Text(label)
+                    .font(Typography.body.font)
+                    .foregroundColor(textColor)
+                Spacer()
+            }
+            .onTapGesture {
+                if state != .disabled {
+                    action()
+                }
+            }
+        }
+        
+        private var borderColor: Color {
+            switch (state, isSelected) {
```

Comment:
Uuuuh...clever...I like it ❤️ I always forget about the "multi variable" `switch`